fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/physical/PhysicalHashAggregateOperator.java

```diff
@@ -124,4 +127,31 @@ public ColumnRefSet getUsedColumns() {
         aggregations.values().forEach(d -> set.union(d.getUsedColumns()));
         return set;
     }
+
+    @Override
+    public boolean couldApplyStringDict(Set<Integer> childDictColumns) {
+        Preconditions.checkState(!childDictColumns.isEmpty());
+        ColumnRefSet dictSet = new ColumnRefSet();
+        for (Integer id : childDictColumns) {
+            dictSet.union(id);
+        }
+
+        for (CallOperator operator : aggregations.values()) {
+            if (!couldApplyStringDict(operator, dictSet)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private boolean couldApplyStringDict(CallOperator operator, ColumnRefSet dictSet) {
+        ColumnRefSet usedColumns = operator.getUsedColumns();
+        if (usedColumns.isIntersect(dictSet)) {
```

Comment:
should check `usedColumns.size() == 1` ?

Reply:
Check  usedColumns size is redundant. because  we  will  check the aggregate  function name, the used column size for one aggregate function is definite